client/tests/AssignTesterDropdown.test.jsx

```diff
@@ -0,0 +1,193 @@
+    } else if (mutation === REMOVE_TESTER_MUTATION) {
+        response = 'REMOVE';
+    } else if (mutation === SCHEDULE_COLLECTION_JOB_MUTATION) {
+        response = 'SCHEDULE';
+    }
+
+    return [jest.fn(() => response), { loading: false, error: null }];
+});
+
+// Mocked GraphQL responses
+const mocks = [
+    {
+        request: {
+            query: ASSIGN_TESTER_MUTATION,
+            variables: {
+                testReportId: 'report1',
+                testerId: '1'
+            }
+        },
+        result: {
+            data: {
+                testPlanReport: {
+                    assignTester: {
+                        testPlanReport: {
+                            draftTestPlanRuns: [
+                                {
+                                    initiatedByAutomation: false,
+                                    tester: {
+                                        id: '1',
+                                        username: 'john'
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+    },
+    {
+        request: {
+            query: REMOVE_TESTER_MUTATION,
+            variables: {
+                testReportId: 'report1',
+                testerId: '1'
+            }
+        },
+        result: {
+            data: {
+                testPlanReport: {
+                    deleteTestPlanRun: {
+                        testPlanReport: {
+                            draftTestPlanRuns: []
+                        }
+                    }
+                }
+            }
+        }
+    },
+    {
+        request: {
+            query: SCHEDULE_COLLECTION_JOB_MUTATION,
+            variables: {
+                testPlanReportId: 'report1'
+            }
+        },
+        result: {
+            data: {
+                scheduleCollectionJob: {
+                    id: 'some-job-id',
+                    status: 'pending'
+                }
+            }
+        }
+    }
+];
+
+describe('AssignTesterDropdown', () => {
+    beforeEach(() => {
+        cleanup();
+    });
+
+    it('renders without crashing', () => {
+        render(
+            <MockedProvider mocks={mocks}>
+                <AssignTesterDropdown {...mockProps} />
+            </MockedProvider>
+        );
+        expect(screen.getByLabelText('Assign testers')).toBeInTheDocument();
+    });
+
+    it('assigns tester correctly and calls assignTester mutation', async () => {
+        render(
+            <MockedProvider mocks={mocks} addTypename={false}>
+                <AssignTesterDropdown {...mockProps} />
+            </MockedProvider>
+        );
+
+        const button = await screen.getByRole('button', {
+            name: /assign testers/i
+        });
+        fireEvent.click(button);
+
+        const items = await screen.findAllByText(/john/);
+        fireEvent.click(items[0]);
```

Comment:
Asserting the value of `items.length` here will make diagnosing failure easier when no element is present (and also prevent false positives if a future change causes multiple items to be rendered).

```suggestion
        const items = await screen.findAllByText(/john/);
        expect(items.length).toBe(1);
        fireEvent.click(items[0]);
```